rio/frontend/policy.ml

```diff
@@ -15,24 +15,47 @@ let lookup s x =
   | Some v -> v
   | None -> raise (UnboundVariable x)
 
+let rec sub_set cl st (p : Ast.set) used =
+  match p with
+  | Class c ->
+      if List.mem c !used then raise (DuplicateClass c)
+      else if List.mem c cl then (
+        used := c :: !used;
+        [ Class c ])
+      else raise (UndeclaredClass c)
+  | Union slst ->
+      let subbed = List.map (fun x -> sub_set cl st x used) slst in
+      List.flatten subbed
+
 let rec sub cl st (p : Ast.policy) used =
   let sub_plst cl st = List.map (fun x -> sub cl st x used) in
   let sub_weighted_plst cl st =
     List.map (fun (x, i) -> (sub cl st x used, i))
   in
 
+  (* Temporary compilation removes FIFOs for test cases *)
+  let rec extract_subpol (p : t) =
```

Comment:
Was this hack to get the `Policy.to_string` tests to pass? 

Also, I totally agree that we should do this though: the new Rio AST `rr[fifo[A], fifo[B]]` is definitely closer to the old AST `rr[A, B]` than `rr[fifo[A], fifo[B]]`.

Reply:
Yeah, just to get the tests to pass. It gets rid of all the 'seemingly unnecessary' FIFOs.
It's perhaps more fitting with the Policy type, so that's fine as is, I think.